octopse.py

```diff
@@ -135,16 +134,17 @@ def deploy_local(data):
             invoke_deploy("{0}\{1}.{2}\PostDeploy.ps1".format(staging, package, version))
 
 
-
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
+    parser.add_argument('infile', nargs='?', type=argparse.FileType('r'), default=sys.stdin)
     parser.add_argument('-e', '--environment', default="local", type=str,
                         help="Supply the environment that you would like to deploy to.")
     parser.add_argument('--wait', action="store_true",
                         help="Wait and poll for deploy")
     parser.add_argument('--force', action='store_true',
                         help='Force deployment')
     args = parser.parse_args()
+    manifest = args.infile

```

Comment:
What if `args.infile` is not supplied? or is ""

Reply:
Currently this will hang i.e. wait for manual input, which seems appropriate? 

Not sure how you'd do anything different/differentiate between piped input vs typed input vs no input.